scripts/notebook_text_checker.py

```diff
@@ -517,6 +517,18 @@ def file_list(path, errors):
 
 This check can be run locally, via `{command}`."""
 
+        if "GITHUB_ACTIONS" in os.environ:
+
+            def github_file_list(path, errors):
+                whole_list = "%0A".join(error.replace("\n", "%0A") for error in errors)
+                return (
+                    f"::error file={path}::Notebook failed text check:%0A{whole_list}"
+                )
+
+            annotation_output = "\n".join(
+                github_file_list(path, errors) for path, errors in all_errors
+            )
+            print(annotation_output)
```

Comment:
I think this can be slightly simplified to a single loop that prints line-by-line?

```python
for path, errors in all_errors:
    whole_list = ...
    message = f"..."
    print(f"::error file={path}::{message}")
```